### DISCUSSION TITLE
PHP Support for Nx Project Graph

### DISCUSSION THREAD

**Discussion by damienwebdev** [nrwl/nx#7841] [Show and Tell]
Hello all,

 I've been using Nx in a variety of non-js projects for a while now (Helm, Markdown Generation, etc). I recently came across the [Project Graph article](https://nx.dev/l/a/structure/project-graph-plugins) and I was curious about the direction that `nrwl` an/or `Nx` will take in the long term in regards to non-js projects.
 
  I'm wondering whether or not it makes sense to start a sidecar-like project for Nx for the PHP ecosystem and what they would look like in terms of interaction with/promotion for/discussions about Nx.
 
 Currently, I'm attempting to target [Magento](https://github.com/magento/magento2) as a possible build target for Nx, as it's a large interdependent PHP package framework that suffers from miserable build performance and could use a little only-build-whats-changed-in-the-dep-graph love.
 
 I've built [a mini-demo here](https://github.com/damienwebdev/demo-nx-php), but I would love some advisement from @vsavkin on where to go next.
 
 A little blessing from the Nx team would go a long way in making clear that this is either "a good idea" or "something that's cool but we're really not going to put any thought or effort into".
 
Thanks,
Damien

**Comment by jackmorizo** [comment]
I think a proper composer integration with nrwl nx would not only be a "good idea" .. it would be awesome.
I know PHP gets a lot of hate but it has awesome features thanks to the efforts of php-fig it's ecosystem is actually quite nice.
